### PR TITLE
[jax2tf] Document the handling of custom JAX pytrees

### PR DESCRIPTION
**Pull request by gnecula**
Add a test also.

### CODE REVIEW

**Comment by gnecula** [comment]
@rxsang PTAL

File: jax/experimental/jax2tf/README.md [hunk 1]
```diff
@@ -708,6 +708,81 @@ g_jaxx2tf_0 = tape.gradient(res, xs,
 # In this case we get the same result as for TF native.
 ```
 
+### Functions whose arguments and results are Python nested data structures
+
+jax2tf can convert functions with arguments and results that are nested
+collections (tuples, lists, dictionaries) of numeric values or JAX arrays. The
```
Comment (rxsang) [inline jax/experimental/jax2tf/README.md:714]: Do you want to just reference to PyTree definition? https://jax.readthedocs.io/en/latest/pytrees.html, or there are limitations to support generic PyTrees?

File: jax/experimental/jax2tf/README.md [hunk 2]
```diff
@@ -708,6 +708,81 @@ g_jaxx2tf_0 = tape.gradient(res, xs,
...
+leaves can be numeric values or TensorFlow tensors (`tf.Tensor`, `tf.TensorSpec`, `tf.Variable`).
+
+As long as the arguments use only standard Python containers (tuple, list, dictionaries),
+both JAX and TensorFlow can flatten and unflatten them and you can use the converted
+function in TensorFlow without limitations.
+
+However, if your JAX function takes a custom container, you can register it with
+the JAX `tree_util` module so that JAX will know how to operate with it, and you
+can still convert the function to use it in TensorFlow
+eager and compiled mode, but you won't be able to save it to a SavedModel, nor
```
Comment (rxsang) [inline jax/experimental/jax2tf/README.md:725]: compiled ->`tf.function`

File: jax/experimental/jax2tf/README.md [hunk 3]
```diff
@@ -708,6 +708,81 @@ g_jaxx2tf_0 = tape.gradient(res, xs,
...
+
+x = CustomPair(4., 5.)
+res_jax = f_jax(x)
+# TF execution works as long as JAX can flatten the arguments
+res_tf = jax2tf.convert(f_jax)(x)
+self.assertAllClose(res_jax, res_tf.numpy())
+res_tf_2 = tf.function(jax2tf.convert(f_jax), autograph=False, jit_compile=True)(x)
+```
+
+If you want to save the function or compute gradients, you should construct
```
Comment (rxsang) [inline jax/experimental/jax2tf/README.md:750]: save the function into SavedModel

**Review by rxsang — APPROVED** [review]

File: jax/experimental/jax2tf/README.md (hunk 1 above)
Comment (gnecula) [reply jax/experimental/jax2tf/README.md:714]: Done

File: jax/experimental/jax2tf/README.md (hunk 3 above)
Comment (gnecula) [reply jax/experimental/jax2tf/README.md:750]: done
